components/ContributionConfirmationModal.js

```diff
@@ -0,0 +1,237 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { getCurrencySymbol } from '../lib/currency-utils';
+import { i18nGraphqlException } from '../lib/errors';
+import { API_V2_CONTEXT, gqlV2 } from '../lib/graphql/helpers';
+
+import Container from './Container';
+import FormattedMoneyAmount from './FormattedMoneyAmount';
+import { Box, Flex } from './Grid';
+import InputField from './InputField';
+import StyledButton from './StyledButton';
+import StyledHr from './StyledHr';
+import Modal, { ModalBody, ModalFooter, ModalHeader } from './StyledModal';
+import { P, Span } from './Text';
+import { TOAST_TYPE, useToasts } from './ToastProvider';
+
+const confirmContributionMutation = gqlV2/* GraphQL */ `
+  mutation ConfirmContribution($id: String!, $action: ProcessOrderAction!, $details: OrderDetailsInput) {
+    processPendingOrder(order: { id: $id }, action: $action, details: $details) {
+      id
+      status
+      permissions {
+        id
+        canMarkAsPaid
+        canMarkAsExpired
+      }
+      amount {
+        currency
+        valueInCents
+      }
+      platformContributionAmount {
+        currency
+        valueInCents
+      }
+    }
+  }
+`;
+
+const ContributionConfirmationModal = ({ order, onClose, show }) => {
+  const amountInitiated = order.amount.valueInCents + order.platformContributionAmount.valueInCents;
```

Comment:
Can we migrate to `platformTipAmount` instead of `platformContributionAmount` please?

Other than that, this is the right calculation for `amountInitiated`.

Reply:
I've change this everywhere except for what is returned by the mutation; 

https://github.com/opencollective/opencollective-frontend/blob/6d7f9fe882ee7e9ce2fb20f6d4d369cc2eb05c5e/components/ContributionConfirmationModal.js#L33-L36

TBC, but I think we need to return the `platformContributionAmount` as this is what is used by the `OrderBudgetItem` to display the platform tip on the `OrderList`. And if we don't return this it will not sync the `OrderList`'s platform tip amount properly after the user completes the modal.

https://github.com/opencollective/opencollective-frontend/blob/b669717e5452e2cf7f71491405f3bfdfaa49d0d6/components/budget/OrderBudgetItem.js#L140-L157

I'll run a test on this once the API PR is completed and verify. 🤔

Reply:
Yes, so I've tested and we have to keep the `platformContributionAmount` as the graphql return so that the graphql cache is correctly updated. But everything else I've used `platformTipAmount`.

Reply:
Why not migrate everything to `platformTipAmount` ? Now or follow up issue?

Reply:
Yes, thanks for reminding. I was thinking of creating a separate issue for this but somehow forgot. I've created this at, https://github.com/opencollective/opencollective/issues/5052. I'll do this today. 👍🏽